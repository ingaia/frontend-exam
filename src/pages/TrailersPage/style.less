@import './../../variables.less';

.trailers-page {
    display: flex;
    height: 100vh;
    width: 100%;
    background: linear-gradient(rgba(20,20,20, .7), rgba(20,20,20, .7)), url("../../../assets/background.png");
    background-size: cover;
}

@media @mobile {
    .trailers-page {
        flex-direction: column;
        height: 100%;
    }
}