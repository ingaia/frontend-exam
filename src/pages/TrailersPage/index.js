import React, { useEffect, useState } from 'react';
import Menu from '../../components/Menu';
import SideMenu from '../../components/SideMenu';
import Trailers from '../../components/Trailers';

import api from '../../services/youtube-api';
import './style.less';

export default function TrailersPage({ history }) {
    const [loading, setLoading] = useState(false);
    const [trailers, setTrailers] = useState([]);
    const [nextPageToken, setNextPageToken] = useState('');

    async function loadPlaylist(params) {
        const response = await api.get('/playlistItems', {
            params
        });
        
        let newTrailers = trailers.concat(response.data.items);
        
        setLoading(false);
        setTrailers(newTrailers);
        setNextPageToken(response.data.nextPageToken);
    }
    
    function loadMore(){
        setLoading(true);
        const params = {
            key: 'AIzaSyBh99ay4rru9-e4pccxqd8BXvPT76Nw4WQ',
            part: 'snippet',
            playlistId: 'PL6t93nUFQQ1ZiXMfhPyhjb0PX3LgEVMcF',
            pageToken: nextPageToken
        }
        loadPlaylist(params);
    }

    useEffect(() => {
        const params = {
            key: 'AIzaSyBh99ay4rru9-e4pccxqd8BXvPT76Nw4WQ',
            part: 'snippet',
            playlistId: 'PL6t93nUFQQ1ZiXMfhPyhjb0PX3LgEVMcF'
        }

        loadPlaylist(params);
    }, []);

    function logOut(){
        history.push('/');
    }

    return (
        <div className="trailers-page">
            <Menu logOut={logOut}/>
            <SideMenu logOut={logOut}/>
            <Trailers trailers={trailers} nextPageToken={nextPageToken} loadMore={loadMore} loading={loading}/>
        </div>
    );
}