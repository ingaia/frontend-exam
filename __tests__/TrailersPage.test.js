import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import TrailersPage from '../src/pages/TrailersPage';

describe('<TrailersPage/>', () => {
    let wrapper;

    beforeEach(() => {

        wrapper = shallow(<TrailersPage/>);
    });

    it('should render', () => {

        expect(wrapper.hasClass('trailers-page')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});