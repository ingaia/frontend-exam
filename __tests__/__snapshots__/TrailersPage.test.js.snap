// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrailersPage/> should render 1`] = `
<div
  className="trailers-page"
>
  <Menu
    logOut={[Function]}
  />
  <SideMenu
    logOut={[Function]}
  />
  <Trailers
    loadMore={[Function]}
    loading={false}
    nextPageToken=""
    trailers={Array []}
  />
</div>
`;
